cluster/metrics/ema.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metrics
+
+import (
+	"fmt"
+	"sync"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common"
+)
+
+// EMA is a struct implemented Exponential Moving Average.
+// val = old * (1 - alpha) + new * alpha
+type EMA struct {
```

Comment:
EMA and silding window counter as metrics names might make others confused. Please give your approach a name, such as "hill climbing" (just an example), and move all of them to `<Name>Metrics`. If it will be extended to more approaches in the future, then they will be named `<Name1>Metrics`, `<Name2>Metrics`, etc.

Reply:
To make more clear, please keep `EMA` and `SlidingWindowCounter` in each file, create a new file, named `<Name>_metrics.go`, then merge `EMAMetrics` and `SlidingWindowCounterMetrics` into `<Name>Metrics`.